catalyst/core/callbacks/pruning.py

```diff
@@ -0,0 +1,186 @@
+from typing import Callable, List, Optional, Union
+import warnings
+
+import torch
+from torch.nn.utils import prune
+
+from catalyst.core import Callback, CallbackOrder, IRunner
+from catalyst.utils.initialization import weight_reset
+
+PRUNING_FN = {
+    "l1_unstructured": prune.l1_unstructured,
+    "random_unstructured": prune.random_unstructured,
+    "ln_structured": prune.ln_structured,
+    "random_structured": prune.random_structured,
+}
```

Comment:
you can just use prune.__dict__[{fn_name_here}]

Reply:
`torch.nn.utils.prune` module also has `global` pruning type methods, which has different APIs. I don't think that putting different behaviour in one callback is a good idea, despite the fact that basically they do the same thing.
Here are examples:
`prune.random_unstructured(module, name="weight", amount=0.3)`
`prune.ln_structured(module, name="weight", amount=0.5, n=2, dim=0)  # we also need dim`
There is almost no difference.
And here is example with `global` pruning:
```python
parameters_to_prune = (
    (model.conv1, 'weight'),
    (model.conv2, 'weight'),
    (model.fc1, 'weight'),
    (model.fc2, 'weight'),
    (model.fc3, 'weight'),
)

prune.global_unstructured(
    parameters_to_prune,
    pruning_method=prune.L1Unstructured,
    amount=0.2,
)
```
So the question is: Do we need `GlobalPruningCallback` or I should integrate global pruning methods to existing callback?

Reply:
Also as you can see global pruning method a kind of more high-level as it is using other pruning_fn

Reply:
🤔  looks like good option for todo